<a href="https://colab.research.google.com/github/Ahowardjr/MSSP6070/blob/main/WeeklyModules/Week%204/Copy_of_Assignment_1_Academic_Performance.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
from google.colab import files
uploaded = files.upload()

Saving data_academic_performance.csv to data_academic_performance (1).csv


In [ ]:
from google.colab import files
uploaded = files.upload()

In [ ]:
import pandas as pd
df = pd.read_csv("data_academic_performance.csv")
df.head()

,COD_S11,GENDER,EDU_FATHER,EDU_MOTHER,OCC_FATHER,OCC_MOTHER,STRATUM,SISBEN,PEOPLE_HOUSE,Unnamed: 9,...,CC_PRO,ENG_PRO,WC_PRO,FEP_PRO,G_SC,PERCENTILE,2ND_DECILE,QUARTILE,SEL,SEL_IHE
0,SB11201210000129,F,Incomplete Professional Education,Complete technique or technology,Technical or professional level employee,Home,Stratum 4,It is not classified by the SISBEN,Three,NaN,...,71,93,79,181,180,91,5,4,2,2
1,SB11201210000137,F,Complete Secundary,Complete professional education,Entrepreneur,Independent professional,Stratum 5,It is not classified by the SISBEN,Three,NaN,...,86,98,78,201,182,92,5,4,4,4
2,SB11201210005154,M,Not sure,Not sure,Independent,Home,Stratum 2,Level 2,Five,NaN,...,18,43,22,113,113,7,1,1,1,1
3,SB11201210007504,F,Not sure,Not sure,Other occupation,Independent,Stratum 2,It is not classified by the SISBEN,Three,NaN,...,76,80,48,137,157,67,4,3,2,2
4,SB11201210007548,M,Complete professional education,Complete professional education,Executive,Home,Stratum 4,It is not classified by the SISBEN,One,NaN,...,98,100,71,189,198,98,5,4,4,2


In [ ]:
df.shape

(12411, 45)

In [ ]:
for column in df. columns:
  print(column)

COD_S11
GENDER
EDU_FATHER
EDU_MOTHER
OCC_FATHER
OCC_MOTHER
STRATUM
SISBEN
PEOPLE_HOUSE
Unnamed: 9
INTERNET
TV
COMPUTER
WASHING_MCH
MIC_OVEN
CAR
DVD
FRESH
PHONE
MOBILE
REVENUE
JOB
SCHOOL_NAME
SCHOOL_NAT
SCHOOL_TYPE
MAT_S11
CR_S11
CC_S11
BIO_S11
ENG_S11
Cod_SPro
UNIVERSITY
ACADEMIC_PROGRAM
QR_PRO
CR_PRO
CC_PRO
ENG_PRO
WC_PRO
FEP_PRO
G_SC
PERCENTILE
2ND_DECILE
QUARTILE
SEL
SEL_IHE


In [ ]:
print(df["EDU_MOTHER"].value_counts(dropna=False))
print(df["EDU_FATHER"].value_counts(dropna=False))
print(df["G_SC"].describe())

EDU_MOTHER
Complete Secundary                       3106
Complete professional education          3059
Complete technique or technology         1495
Incomplete Secundary                     1056
Postgraduate education                    997
Complete primary                          713
Incomplete primary                        541
Incomplete Professional Education         502
0                                         388
Incomplete technical or technological     341
Not sure                                  179
Ninguno                                    34
Name: count, dtype: int64
EDU_FATHER
Complete professional education          3016
Complete Secundary                       2843
Complete technique or technology         1194
Incomplete Secundary                     1091
Postgraduate education                   1085
Complete primary                          824
Incomplete primary                        735
Incomplete Professional Education         425
Not sure                        

In [ ]:
df.groupby("EDU_MOTHER")["G_SC"].agg(["count", "mean"]).sort_values("mean")

,count,mean
EDU_MOTHER,,
Ninguno,34,149.705882
Incomplete primary,541,154.853974
0,388,154.930412
Complete primary,713,155.336606
Incomplete Secundary,1056,155.672348
Complete Secundary,3106,158.470702
Not sure,179,159.407821
Incomplete technical or technological,341,161.879765
Complete technique or technology,1495,163.242140


In [ ]:
df.groupby("EDU_FATHER")["G_SC"].agg(["count", "mean"]).sort_values("mean")

,count,mean
EDU_FATHER,,
0,391,155.076726
Complete primary,824,155.195388
Incomplete primary,735,155.729252
Incomplete Secundary,1091,156.535289
Ninguno,123,156.634146
Complete Secundary,2843,158.785790
Not sure,407,160.294840
Incomplete technical or technological,277,162.101083
Complete technique or technology,1194,164.033501


In [ ]:
mother_means = df.groupby("EDU_MOTHER")["G_SC"].mean()
father_means = df.groupby("EDU_FATHER")["G_SC"].mean()

# Identify the exact keys from the series indices to avoid KeyError due to subtle string differences
mother_primary_key = next((k for k in mother_means.index if k.strip().lower() == 'complete primary'), None)
mother_postgraduate_key = next((k for k in mother_means.index if k.strip().lower() == 'postgraduate education'), None)

father_primary_key = next((k for k in father_means.index if k.strip().lower() == 'complete primary'), None)
father_postgraduate_key = next((k for k in father_means.index if k.strip().lower() == 'postgraduate education'), None)

# Add checks for None in case keys are truly missing
if mother_primary_key is None:
    raise KeyError("Could not find 'Complete primary' in mother's education categories.")
if mother_postgraduate_key is None:
    raise KeyError("Could not find 'Postgraduate education' in mother's education categories.")
if father_primary_key is None:
    raise KeyError("Could not find 'Complete primary' in father's education categories.")
if father_postgraduate_key is None:
    raise KeyError("Could not find 'Postgraduate education' in father's education categories.")

mother_difference = (
    mother_means[mother_postgraduate_key]
    - mother_means[mother_primary_key]
)
father_difference = (
    father_means[father_postgraduate_key]
    - father_means[father_primary_key]
)

print(f"Mother's education difference: {mother_difference:.2f} points")
print(f"Father's education difference: {father_difference:.2f} points")

Mother's education difference: 20.30 points
Father's education difference: 21.79 points


Does parental education influence outcomes, and if so how much?
I compared students’ average scores by their mothers’ and fathers’ education levels. Students whose mothers had postgraduate education scored 20.30 points higher, on average, than students whose mothers had completed primary school. The corresponding difference for fathers’ education was 21.79 points. These results suggest that higher parental education is associated with higher student scores, although the comparison does not establish that parental education caused the difference.

# 2. Which groups/stratum performs best? Are there differences between groups/stratums? Explain your analysis.


In [ ]:
df.groupby("STRATUM")["G_SC"].agg(["count", "mean", "std"]).sort_values("mean", ascending = False).round(2)

,count,mean,std
STRATUM,,,
Stratum 6,403,181.51,21.88
Stratum 5,633,177.39,21.54
Stratum 4,1578,172.00,21.47
Stratum 3,4045,163.82,21.69
Stratum 2,4029,158.20,22.28
0,14,154.00,21.32
Stratum 1,1709,152.35,21.94


In [ ]:
stratum_means = df.groupby("STRATUM")["G_SC"].mean()
stratum_means["Stratum 6"] - stratum_means["Stratum 1"]

np.float64(29.15832833619126)

In [ ]:
stratum_means = df[df["STRATUM"].isin(
    [f"Stratum {i}" for i in range(1, 7)]
)].groupby("STRATUM")["G_SC"].mean()

stratum_means = stratum_means.reindex(
    [f"Stratum {i}" for i in range(1, 7)]
)

comparison = pd.DataFrame({
    "average_score": stratum_means.round(2),
    "difference_from_previous": stratum_means.diff().round(2)
})
comparison

,average_score,difference_from_previous
STRATUM,,
Stratum 1,152.35,NaN
Stratum 2,158.20,5.85
Stratum 3,163.82,5.62
Stratum 4,172.00,8.19
Stratum 5,177.39,5.38
Stratum 6,181.51,4.12


#2 . Which groups/stratum performs best? Are there differences btween group startums. Above we see startum 6 has the highest average scores, ultimately performing the best. Average scores increased from Stratum 1 to Stratum 6, and there were increases in each level ranging from 4.12 to 8.19, wih Stratum 3-4 having the highest increase. The total difference in scores was 29.16.

# 3. How does each gender perform in the global assessment?


In [ ]:
df.groupby("GENDER")["G_SC"].agg(["count", "mean", "std"]).round (2)

,count,mean,std
GENDER,,,
F,5043,161.28,22.33
M,7368,163.69,23.58


In [ ]:
gender_means = df.groupby("GENDER")["G_SC"].mean()
gender_means["M"]- gender_means["F"]

np.float64(2.4126177737903447)

# 3. On avergae, male student scored 163.69, compared with an average of 161.27 for female students. Therefore, in this assemessment, male students avergae 2.41 more points than female students in this assessment.

4. Use your analysis to understand the results.  What did you find interesting, and can you make any recommendations or inferences?

What stood out most in my analysis was the gap between students in different strata. The average global score was about 29 points higher in Stratum 6 than in Stratum 1. Parental education showed a similar pattern: students whose parents had postgraduate education scored about 20–22 points higher than students whose parents had completed primary education. There wasn’t a large difference by gender, however, as male students scored only about 2.4 points higher on average than female students.

Since the dataset includes social and economic factors, I think it would be interesting to look at the resources available to students in each group. That might help explain the differences in scores, though we would need more analysis before saying what caused them.